tests/test_async_fetch.py

```diff
@@ -732,35 +730,39 @@ def __int__(self):
                 'select <decimal>$arg',
                 arg="10.2")
 
-    async def test_async_wait_cancel_01(self):
-        # Test that client protocol handles waits interrupted
-        # by closing.
-        lock_key = tb.gen_lock_key()
-
-        con2 = await self.connect(database=self.con.dbname)
-
-        await self.con.query_one(
-            'select sys::advisory_lock(<int64>$0)',
-            lock_key)
-
-        try:
-            async with tg.TaskGroup() as g:
-
-                async def exec_to_fail():
-                    with self.assertRaises(ConnectionAbortedError):
-                        await con2.query(
-                            'select sys::advisory_lock(<int64>$0)', lock_key)
-
-                g.create_task(exec_to_fail())
-
-                await asyncio.sleep(0.1)
-                await con2.aclose()
-
-        finally:
-            self.assertEqual(
-                await self.con.query(
-                    'select sys::advisory_unlock(<int64>$0)', lock_key),
-                [True])
+    # Advisory locks don't work/ for now
+    # Can't xfail because:
+    #     database "asyncfetch" is being accessed by other users
+    #
+    # async def test_async_wait_cancel_01(self):
+    #     # Test that client protocol handles waits interrupted
+    #     # by closing.
+    #     lock_key = tb.gen_lock_key()
+    #
+    #     con2 = await self.connect(database=self.con.dbname)
+    #
+    #     await self.con.query_one(
+    #         'select sys::advisory_lock(<int64>$0)',
+    #         lock_key)
+    #
+    #     try:
+    #         async with tg.TaskGroup() as g:
+    #
+    #             async def exec_to_fail():
+    #                 with self.assertRaises(ConnectionAbortedError):
+    #                     await con2.query(
+    #                         'select sys::advisory_lock(<int64>$0)', lock_key)
```

Comment:
Yeah, we test it that way because using sleep() is unreliable. Making them testmode-only is OK.

Reply:
The only reliable way to use them is in transaction, since connection pool is implemented.

Anyway, if we move them to testmode, does this mean that we can also move sleep to testmode? As you said that the main motivation of keeping sleep is to use them in tests of the bindings.